plugins/inputs/bigbluebutton/bigbluebutton.go

```diff
@@ -0,0 +1,217 @@
+package bigbluebutton
+
+import (
+	"crypto/sha1"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/common/proxy"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type BigBlueButton struct {
+	URL              string `toml:"url"`
+	PathPrefix       string `toml:"path_prefix"`
+	SecretKey        string `toml:"secret_key"`
+	getMeetingsURL   string
+	getRecordingsURL string
+
+	tls.ClientConfig
+	proxy.HTTPProxy
+	client *http.Client
+}
+
+var defaultPathPrefix = "/bigbluebutton"
+
+var bbbConfig = `
+	## Required BigBlueButton server url
+	url = "http://localhost:8090"
+
+	## BigBlueButton path prefix. Default is "/bigbluebutton"
+	# path_prefix = "/bigbluebutton"
+
+	## Required BigBlueButton secret key
+	# secret_key =
+
+	## Optional HTTP Proxy support
+	# http_proxy_url = ""
+
+	## Optional TLS Config
+	# tls_ca = "/etc/telegraf/ca.pem"
+	# tls_cert = "/etc/telegraf/cert.pem"
+	# tls_key = "/etc/telegraf/key.pem"
+	## Use TLS but skip chain & host verification
+	# insecure_skip_verify = false
+`
+
+func (b *BigBlueButton) Init() error {
+	if b.SecretKey == "" {
+		return fmt.Errorf("BigBlueButton secret key is required")
+	}
+
+	if b.PathPrefix == "" {
+		b.PathPrefix = defaultPathPrefix
+	}
+
+	b.getMeetingsURL = b.getURL("getMeetings")
+	b.getRecordingsURL = b.getURL("getRecordings")
+
+	tlsCfg, err := b.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	proxy, err := b.HTTPProxy.Proxy()
+	if err != nil {
+		return err
+	}
+
+	transport := &http.Transport{
+		TLSClientConfig: tlsCfg,
+		Proxy:           proxy,
+	}
+
+	b.client = &http.Client{
+		Transport: transport,
```

Comment:
Do you also want to allow the user to specify a timeout for the requests? This is helpful in cases where the request hangs otherwise.